api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTests.java

```diff
@@ -0,0 +1,256 @@
+        Iterable<Claimant> claims = claimantRepository.findAll();
+        assertThat(claims).hasSize(1);
+        Claimant persistedClaim = claims.iterator().next();
+        assertClaimantMatchesClaimantDTO(claimantDTO, persistedClaim);
+        assertThat(persistedClaim.getEligibilityStatus()).isEqualTo(eligibilityStatus);
+    }
+
+    @Test
+    void shouldFailWhenEligibilityServiceCallThrowsException() {
+        //Given
+        ClaimDTO claim = aValidClaimDTO();
+        given(restTemplateWithIdHeaders.postForEntity(anyString(), any(), eq(EligibilityResponse.class))).willThrow(new RestClientException("Test exception"));
+        //When
+        ResponseEntity<ErrorResponse> response = restTemplate.exchange(buildRequestEntity(claim), ErrorResponse.class);
+        //Then
+        assertErrorResponse(response, "An internal server error occurred", INTERNAL_SERVER_ERROR);
+        assertClaimantPersistedSuccessfully(claim.getClaimant(), EligibilityStatus.ERROR);
+        verify(restTemplateWithIdHeaders).postForEntity("http://localhost:8100/v1/eligibility", aValidPerson(), EligibilityResponse.class);
+    }
+
+    @Test
+    void shouldReturn404ErrorForNonExistentPath() {
+        //When
+        ResponseEntity<ErrorResponse> response = restTemplate.getForEntity("/missing-resource", ErrorResponse.class);
+        //Then
+        assertThat(response.getStatusCode()).isEqualTo(NOT_FOUND);
+    }
+
+    @ParameterizedTest(name = "Field {0} with invalid value {1} on a claim returns the correct error response")
+    @MethodSource("shouldFailWithValidationErrorArguments")
+    void shouldFailWithValidationError(String fieldName, String value, String expectedErrorMessage, String expectedField) {
+        //Given
+        ClaimDTO claim = aValidClaimDTO();
+        modifyFieldOnObject(claim.getClaimant(), fieldName, value);
+        //When
+        ResponseEntity<ErrorResponse> response = restTemplate.exchange(buildRequestEntity(claim), ErrorResponse.class);
+        //Then
+        assertValidationResponse(response, expectedField, expectedErrorMessage);
+    }
+
+    @ParameterizedTest(name = "Field {0} with invalid value {1} on an address returns the correct error response")
+    @MethodSource("shouldFailWithAddressValidationErrorArguments")
+    void shouldFailWithAddressValidationError(String fieldName, String value, String expectedErrorMessage, String expectedField) {
+        //Given
+        ClaimDTO claim = aValidClaimDTO();
+        modifyFieldOnObject(claim.getClaimant().getCardDeliveryAddress(), fieldName, value);
+        //When
+        ResponseEntity<ErrorResponse> response = restTemplate.exchange(buildRequestEntity(claim), ErrorResponse.class);
+        //Then
+        assertValidationResponse(response, expectedField, expectedErrorMessage);
+    }
+
+    @ParameterizedTest(name = "Field {0} with invalid value {1} on an error response")
+    @MethodSource("shouldFailWithInvalidDateFormatErrorArguments")
```

Comment:
CsvSource might be better here as the test data will be above the test class. Uses less code as well.
E.g.
```
@CsvSource({
            "1999/12/31, dateOfBirth, '1999/12/31' could not be parsed as a LocalDate, claimant.dateOfBirth",
            "Foo, dateOfBirth, 'Foo' could not be parsed as a LocalDate, claimant.dateOfBirth",
            "29-11-1909, expectedDeliveryDate, '29-11-1909' could not be parsed as a LocalDate, claimant.expectedDeliveryDate",
            "1999/12/31, expectedDeliveryDate, '1999/12/31' could not be parsed as a LocalDate, claimant.expectedDeliveryDate",
            "Foo, expectedDeliveryDate, 'Foo' could not be parsed as a LocalDate, claimant.expectedDeliveryDate"
    })
```

Reply:
I think I'd go for the csv option too.

Reply:
Ok, I'm happy with either but I'll change it as you've both commented on it.